internal/olm/operator/registry/index_image.go

```diff
@@ -71,9 +77,25 @@ func (c IndexImageCatalogCreator) CreateCatalog(ctx context.Context, name string
 		return nil, fmt.Errorf("error in getting registry pod: %v", err)
 	}
 
-	// make catalog source the owner of registry pod object
-	if err := controllerutil.SetOwnerReference(cs, pod, c.cfg.Scheme); err != nil {
-		return nil, fmt.Errorf("error in setting registry pod owner reference: %v", err)
+	// get registry pod key
+	podKey := types.NamespacedName{
+		Namespace: c.cfg.Namespace,
+		Name:      pod.Name,
+	}
+	if err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		if err := c.cfg.Client.Get(ctx, podKey, pod); err != nil {
+			return fmt.Errorf("get registry pod: %v", err)
+		}
+		// make catalog source the owner of registry pod object
+		if err := controllerutil.SetOwnerReference(cs, pod, c.cfg.Scheme); err != nil {
+			return fmt.Errorf("error in setting registry pod owner reference: %v", err)
+		}
+		if err := c.cfg.Client.Update(ctx, pod); err != nil {
+			return fmt.Errorf("error in updating catalog source as owner of pod: %v", err)
+		}
+		return nil
+	}); err != nil {
+		return nil, err
```

Comment:
Nit: instead of creating the registry pod and then updating it with the owner reference, can we put the owner reference on before creating it to reduce the number of requests to the API server?

Reply:
Refactor nit: I don't see a good reason to have much of this logic external to `registryPod.Create()`. Creation, setting owner refs, and verifying that the pod is running should be done in that method, much like for `RegistryResources`.

Reply:
thanks for the inputs @joelanford @estroz, I totally agree! I refactored based on the feedback. Please review new changes.